src/highlight/highlightUtil.ts

```diff
@@ -58,3 +58,65 @@ export const centerHighlight = (shapes: Shape[]): Position => {
 export function isHighlight(annotation: Annotation): annotation is AnnotationHighlight {
     return annotation?.target?.type === Type.highlight;
 }
+
+export const dedupeRects = (rects: Shape[], threshold = 0.1): Shape[] => {
+    const dedupedRects: Shape[] = [];
+
+    rects.forEach(curr => {
+        const prev = dedupedRects.pop();
+        // empty list, push current
+        if (!prev) {
+            dedupedRects.push(curr);
+            return;
+        }
+
+        const heightThreshold = Math.max(
+            Math.floor(prev.height * threshold + 0.5),
```

Comment:
Is this `0.5` different from that in `groupByRow`? If so, how? If not, should they use a constant?